phaser/steps.py

```diff
@@ -13,84 +13,161 @@
 CONTEXT_STEP = "CONTEXT_STEP"
 PROBE_VALUE = "__PROBE__"
 
-
-def row_step(step_function):
+class StepWrapper():
```

Comment:
I don't love this on first sight because it seems too clever!  Sometimes reducing repetition in code has too high a cost in complexity.  But I will definitely give it a chance because it might in fact be better.

Reply:
I agree about cleverness often being not worth it.  Take a look at what the repetition looks like in [steps.py](https://github.com/lisad/phaser/blob/43a76183ddd3bb721d5fc31efe7cbbacbdd5f7ed/phaser/steps.py) before this refactor.  What do you think -- worth the cleverness?

Or... do you have an alternative suggestion for a more straightforward refactor that reduces the repetition?

Reply:
It probably is worth the cleverness.  You were probably about to add even more repetition to steps with inputs and outputs, and there's only so much you can do with helper functions which are my basic goto.  Sorry I came off too negative, what I meant was that I don't really have detailed comments but I think we should try this and merge this PR.  I may have comments later or I may learn something!  I've not used a class wrapper before.

Reply:
Oh, I'm also interested in trying it out and I think we should go forward.  Sorry if I came across too negative.  I need to experience this & it's always good to learn something new -

Reply:
I didn’t interpret your comment negatively.  I’m sure there is another way to do this; I’m happy as it is and we can always change in the future.  It is definitely not meant to be pet if the public API, so its structure can change willy nilly.